plugins/connection/podman.py

```diff
@@ -149,10 +153,11 @@ def exec_command(self, cmd, in_data=None, sudoable=False):
 
         # shlex.split has a bug with text strings on Python-2.6 and can only handle text strings on Python-3
         cmd_args_list = shlex.split(to_native(cmd, errors='surrogate_or_strict'))
+        exec_args_list = ["exec"]
         if self.user:
-            cmd_args_list += ["--user", self.user]
+            exec_args_list.extend(("--user", self.user))
 
-        rc, stdout, stderr = self._podman("exec", cmd_args_list, in_data)
+        rc, stdout, stderr = self._podman(exec_args_list, cmd_args_list, in_data)
```

Comment:
I think this is a problem that we're passing a list here instead of a string. `_podman()` expects a string.

Reply:
My bad, I was looking at the Ansible v2.9 code!

Reply:
I looked at this again, and this change actually applies to both Ansible 2.9 and Ansible 2.10+. There are some small differences in `_podman()` between 2.9 and 2.10, but in particular, this commit changes the type of the `cmd` parameter that `_podman()` will accept. I've updated the docstring to make this clear, at https://github.com/containers/ansible-podman-collections/pull/79

Reply:
I've cherry-picked this for Ansible 2.9 here: https://github.com/ansible/ansible/pull/70541

Reply:
@ktdreyer you might want to cherry-pick buildah change too: https://github.com/containers/ansible-podman-collections/commit/1c3d2456b7a5270ff264407ee927074e9ad8961a

Reply:
If it's ok with you, I will leave that up to someone else, because I don't personally use the buildah plugin. I just need this change for Molecule with Ansible 2.9, because I have several playbooks that use "become".

Reply:
@ktdreyer  Sure, it's not something urgent. Can you please point me in your time to example of usage podman connection plugin with molecule? I'm curios about the use cases. 
Thanks

Reply:
Sure. I have [a repository with playbooks](https://github.com/ktdreyer/koji-playbooks) that set up an application for me ([Koji](https://pagure.io/koji)). I use Molecule to set up and test Koji within a podman container.

 I'm using `become: yes` and `become_method: sudo` in my [top-level Molecule playbook](https://github.com/ktdreyer/koji-playbooks/blob/master/molecule/centos8/converge.yml). This means I run all tasks as `root` by default, and then I have several Kerberos-related tasks that I run as as the unprivileged user with `become: no`.

I found that Ansible would always run everything as root, even with `become: no`, because the podman connection plugin was running `podman exec` with the `--user` flag in the incorrect position. With this patch, Ansible 2.9 correctly runs the tasks as the unprivileged user.